crates/cnidarium/src/rpc.rs

```diff
@@ -160,31 +173,43 @@ async fn watch_changes(
         }
         let (version, changes) = changes_rx.borrow_and_update().clone();
 
-        for (key, value) in changes.unwritten_changes().iter() {
-            if key_regex.is_match(key) {
-                tx.send(Ok(WatchResponse {
-                    version,
-                    entry: Some(wr::Entry::Kv(wr::KeyValue {
-                        key: key.clone(),
-                        value: value.as_ref().cloned().unwrap_or_default(),
-                        deleted: value.is_none(),
-                    })),
-                }))
-                .await?;
+        if key_regex.is_some() || nv_key_regex.is_none() {
+            for (key, value) in changes.unwritten_changes().iter() {
+                if key_regex
+                    .as_ref()
```

Comment:
i think the `Option<T>` changes here have ended up convoluting the logic a bit.

above, we match on..

```rust
        let nv_key_regex = match request.nv_key_regex.as_str() {
            "" => None,
            _ => Some(
                regex::bytes::RegexBuilder::new(&request.nv_key_regex)
                    // .. elided
                )
        };
```

..which we pass into this function and subsequently re-map `Option::None` back into an empty regex. it seems like we can both keep this diff smaller, and keep the underlying code the same, if we walk back the change to make these regex paremeters optional values.

what do you think @zbuc?

Reply:
I considered this however my understanding is that we cannot use optional parameters in protobuf as the Golang implementation doesn't handle them correctly. We previously used optional parameters all over, and had to excise them. We could get around this in a couple ways, either by using an enum, or a placeholder string value.

Reply:
hm, i worry i might not have communicated this clearly. let me open a branch against this...

Reply:
see #4364, this is what i was trying to get at. does that seem right to you? apologies if i am missing something here.

Reply:
marking as resolved :white_check_mark: